Index: lib/SPIRV/SPIRVRegularizeLLVM.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/SPIRV/SPIRVRegularizeLLVM.cpp b/lib/SPIRV/SPIRVRegularizeLLVM.cpp
--- a/lib/SPIRV/SPIRVRegularizeLLVM.cpp	(revision 8703d43bd14d000fc630c2d3918d918819d23741)
+++ b/lib/SPIRV/SPIRVRegularizeLLVM.cpp	(revision 44eda7be5314a7c7b1d8a33b72d5961842ca60b4)
@@ -413,6 +413,17 @@
           if (isa<PossiblyExactOperator>(BO) && BO->isExact())
             BO->setIsExact(false);
         }
+
+        // FIXME: This is not valid handling for freeze instruction
+        if (auto *FI = dyn_cast<FreezeInst>(&II)) {
+          auto *V = FI->getOperand(0);
+          if (isa<UndefValue>(V))
+            V = Constant::getNullValue(V->getType());
+          FI->replaceAllUsesWith(V);
+          FI->dropAllReferences();
+          ToErase.push_back(FI);
+        }
+
         // Remove metadata not supported by SPIRV
         static const char *MDs[] = {
             "fpmath",
